### DISCUSSION TITLE
Unable to run the Player on Spyder and VSCode

### DISCUSSION THREAD

**Discussion by rogergallart** [kineticstoolkit/kineticstoolkit#96] [1. Users Q&A] [answered]
Hi
I was trying to start using this library and I tried to run this code I found on the documentation without success:
```python
import kineticstoolkit.lab as ktk
markers = ktk.kinematics.read_c3d_file(
    ktk.doc.download('kinematics_racing_full.c3d')
)
viewing_options = {
    'zoom': 3.5,
    'azimuth': 0.8,
    'elevation': 0.16,
    'translation': (0.2, -0.7)
}
player = ktk.Player(markers, **viewing_options)
player.to_html5(start_time=0, stop_time=10)
```

- when I run it on Spyder I just get an empty window (that can not be closed) and nothing else happens:
<img width="562" alt="Screenshot 2022-05-30 at 11 56 23" src="https://user-images.githubusercontent.com/11130916/170969479-23340a7c-7142-4f2c-a26d-1890696bc4c7.png">

- if I run it on VS Code, nothing happens (not even an error)
- if I run it on Jupyter notebook, adding "%matplotlib qt5" at the beginning, it works. It shows the video as an output. Note that I had to manually install the ffmpeg package (not included as a dependency???)

Can anyone help me to make it work in Spyder and VSCode ? I am using MacOS BigSur (Intel silicon).
Thanks in advance

**Comment by felixchenier** [comment, marked as answer]
Hello @rogergallart 

Thanks for the detailed question. We will narrow down your issue until it works.

First, just to clarify, the last line (`player.to_html5(start_time=0, stop_time=10)`) is not needed. It's only needed for jupyter-based environments (such as the website) and provides only a non-interactive experience. This is the only line that requires ffmpeg, but since it's only useful for Jupyter, I decided to not require it as a dependency (I may revisit this choice someday).

Now, for your issue. Please start an IPython console in the terminal:

```
conda activate `whatever_environment_you_created`
ipython
```

then:

```
%matplotlib qt5
import kineticstoolkit.lab as ktk
markers = ktk.kinematics.read_c3d_file(
    ktk.doc.download('kinematics_racing_full.c3d')
)
ktk.Player(markers)
```

Does the interactive player open?

**Reply by rogergallart** [reply]
Hello @felixchenier,

Thanks for your reply !

Sorry, I read the documentation too fast and did not notice that the "Player.to_html5" line was just for notebooks :-P  

After removing this line, things have improved. I get the points plot in the window and I can move around the scene (zoom, etc), however I can not see the animation (the points are static, not moving).
<img width="598" alt="Screenshot 2022-05-31 at 10 38 27" src="https://user-images.githubusercontent.com/11130916/171130724-c8368738-39c4-479d-9c02-4bf6b3cd3e44.png">

If as you suggested I run the code in the IPython console, I get the same result: the interactive player opens, it shows the points, I can move around the scene but the animation is not played. 
**Question: should I see some playback control buttons on the window (play/stop/pause) or not ? I don't see any button but I don't know if it should be this way.**

Also notice that I got this warning:
```
qt.qpa.drawing: Layer-backing can not be explicitly controlled on 10.14 when built against the 10.14 SDK
```
which seems to be related to be using MacOS BigSur. 

Best regards

**Reply by felixchenier** [reply]
Hi @rogergallart 

Nice, I think you're there. I just realized that nowhere I explained how to use the Player on the website. It's now on my todo list.

There is no clickable control, everything is controlled via keypresses or mouse drag/scroll. To get help on the available commands, press `h`. You'll get this nice old-fashioned green-text layer to guide you.

<img width="1012" alt="Screen Shot 2022-05-31 at 07 54 34" src="https://user-images.githubusercontent.com/34967663/171167365-df4eecc6-75d7-4ef0-8e16-ca6a31ab7c01.png">

For the qt warning, it seems to be generalized in many other projects using qt on Big Sur, so I guess I couldn't do anything for it. For what it's worth, I tried the Player on Monterey in a virtual machine and it does not show the warning, but I couldn't be sure this would be the same on a real Monterey machine.

Does this solve your issue?

**Reply by rogergallart** [reply]
It works ! :-D. Thanks
Congratulations for this project. It looks very promising. 
I would like to start contributing to it. Maybe I could start writing the explanation on how to use the Player ?

**Reply by felixchenier** [reply]
@rogergallart Excellent!

A big YES for your desire to contribute to the project :-)

I created an issue for this documentation for the rest of this discussion: #97.

See you in this issue!
